crates/shared/src/sources/balancer_v2/pool_fetching.rs

```diff
@@ -227,6 +208,44 @@ impl Maintaining for BalancerPoolFetcher {
     }
 }
 
+/// Creates an aggregate fetcher for all supported pool factories.
+async fn create_all_pools_fetcher(
```

Comment:
Nit. Might be better to call create all pool fetchers (I.e move the plural from pools to fetcher) since that's what it does.